src/client/features/view/components/tooltips/generateContent.js

```diff
@@ -0,0 +1,218 @@
+const h = require('hyperscript');
+const config = require('../../config');
+
+//Generate HTML elements for a Parsed Metadata Field
+//Optional trim parameter indicates if the data presented should be trimmed to a reasonable length
+//Data Pair -> HTML
+function parseMetadata(pair, trim = true) {
```

Comment:
I think this function can be rewritten using the command pattern.

http://gameprogrammingpatterns.com/command.html

If I understand this function correctly, a pair can have one of many key types, and depending on the key you do different formatting right?

I think it would be better if you have a map for each key type and key type handler.

e.g
```js
const metaDataKeyMap = new Map()
.set('Standard Name', standardNameHandler)
...
const standardNameHandler = (pair) => {
   return makeTooltipItem(...);
}
```

and then parseMetaData can just be:
```js
const parseMetadata = (pair, trim = true) => {
  return metaDataKeyMap.get(pair[0])(pair);
}
```

Reply:
@d2fong  This does seem alot more cleaner to understand/read. 
I have converted all the if blocks into their own individual functions and set them in a map.

Reply:
I see a number of potential benefits of this approach:

1. It makes the reader aware that you do different things based on the meta data key; something that is harder to see in a cascading if statement.
2. It segments the code into more digestible chunks.
3. Makes each chunk more malleable and receptive to change.
4. It is easy to find via function name. E.g if you want to change anything to the old version of the function you have to scroll and look for a key string specifically. In the new version you can just look for the key in the function name itself.